components/core/JsInterop/DomEventListener.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Components;
+using Microsoft.JSInterop;
+
+namespace AntDesign.JsInterop
+{
+    public class DomEventListener<T>
+    {
+        private readonly IJSRuntime _jsRuntime;
+        private Dictionary<string, DotNetObjectReference<Invoker<T>>> _dotNetObjectStore = new();
+
+        public DomEventListener(IJSRuntime jsRuntime)
+        {
+            _jsRuntime = jsRuntime;
+        }
+
+        public void Add(object dom, string eventName, Action<T> callback, bool preventDefault = false)
```

Comment:
Can we use the generic type at `Add` method instead of class name?

Reply:
Because we may use more than one return type at a time.

Reply:
Look like not possible, VS doesn't know what is the generic type for _dotNetObjectStore.

![image](https://user-images.githubusercontent.com/4554555/131294096-ecab2deb-5fd3-4a0b-8858-eb3d45af9f6c.png)

Reply:
Maybe can use `Dictionary<string, IDisposable>`.

BTW, I think we also need to add the Exclusive logic.

Reply:
`Dictionary<string, IDisposable>` works perfectly as expected. Pushed amended code and now with Exclusive and reuse logic in the same class/service.

Reply:
Thank you!

Reply:
I noticed there were failures on the bunit test. Do I need to do something about it?